sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -2313,9 +2407,13 @@ pi_result piEventsWait(pi_uint32 NumEvents, const pi_event *EventList) {
     // NOTE: we are destroying associated command lists here to free
     // resources sooner in case RT is not calling piEventRelease soon enough.
     if (EventList[I]->ZeCommandList) {
-      // Event has been signaled: Destroy the command list associated with the
-      // call that generated the event.
-      ZE_CALL(zeCommandListDestroy(EventList[I]->ZeCommandList));
+      // Event has been signaled: If the fence for the associated command list
+      // is signalled, then reset the fence and command list and add them to the
+      // available list for ruse in PI calls.
+      EventList[I]->Queue->Context->Device->ZeCommandListFenceMapMutex.lock();
```

Comment:
can the lock/unlock go inside the resetCommandListFenceEntry, where they are really needed?

Reply:
so, that would work except for when this is called in getAvailableCommandList. because we don't want recursive locking to be supported this causes a deadlock. This was why I moved the lock to outside this function.

Reply:
OK, I don't like all our locking infrastructure, will rework that separately one day.